api/rest/clientInterface/clientInterface.go

```diff
@@ -0,0 +1,45 @@
+package clientInterface
+
+import (
+	"context"
+
+	"github.com/ipfs/ipfs-cluster/api"
+
+	cid "github.com/ipfs/go-cid"
+	shell "github.com/ipfs/go-ipfs-api"
+	files "github.com/ipfs/go-ipfs-cmdkit/files"
+	peer "github.com/libp2p/go-libp2p-peer"
+)
+
+// ClientIface defines the interface to be used by API clients to interact with
+// the ipfs-cluster-service
+type ClientIface interface {
```

Comment:
We will need to document each of these methods here, if we make the client private. Basically consists on copying the docstrings for each method in the current client to the methods here.